js-packages/@prestojs/viewmodel/src/ViewModelCache.ts

```diff
@@ -243,10 +324,84 @@ class RecordCache<ViewModelClassType extends ViewModelConstructor<any, any, any>
                 }
             }
         }
-
+        this.onAnyChange();
         return ret;
     }
 
+    /**
+     * Adds a placeholder entry for `fieldNames` so that changes to that cache key are detected
+     * (eg. when a superset of fields are changed). Changes to a subset of fields are detected
+     * by their existence in either listener cache or record cache. This handles almost all cases
+     * but as we support traversing related caches there is the case where some part of the chain
+     * legitimately has a null value and so the related record can't be returned but it's still
+     * a valid record. For example if you try to retrieve a record with `get(1, ['user', 'group', 'owner'])`
+     * but the `groupId` is null (ie. group is actually optional) then this should still return
+     * a record as it exists it's just that the relations aren't set. This won't be detected based
```

Comment:
the relation could be null because either a) the related record isn't in the cache b) there's just no relation (ie. the id is set to null)

So I just mean it should be a cache hit - it exists in the cache despite not having those nested fields (ie. it can't have the nested fields because the relation is null)

Reply:
Ah so when you say "the relation could be null", the FK is not null but the RelatedField field is?

Can you clarify that a bit more in the comments -- specifically when you say "the relation" it's not clear that you mean "the RelatedViewModelField field" and not necessarily the FK source field